src/main/java/mops/businesslogic/file/query/FileQuery.java

```diff
@@ -65,15 +65,33 @@ private boolean checkTags(FileInfo file) {
         return tags.isEmpty() || anyMatch;
     }
 
+    @SuppressWarnings("PMD.OnlyOneReturn")
     private boolean checkTypes(FileInfo file) {
-        return types.isEmpty() || types.contains(file.getType().toLowerCase(Locale.ROOT));
+        for (String type : types) {
+            if (file.getType().toLowerCase(Locale.ROOT).contains(type.toLowerCase(Locale.ROOT))) {
+                return true;
+            }
+        }
+        return types.isEmpty();
     }
 
+    @SuppressWarnings("PMD.OnlyOneReturn")
     private boolean checkNames(FileInfo file) {
-        return names.isEmpty() || names.contains(file.getName().toLowerCase(Locale.ROOT));
+        for (String name : names) {
+            if (file.getName().toLowerCase(Locale.ROOT).contains(name.toLowerCase(Locale.ROOT))) {
+                return true;
+            }
+        }
```

Comment:
Couldn't that be handled in a stream?

Reply:
Streams are annoying. How exactly would the stream look? If it works like this I see no reason to change it and invest more time in a working feature before completing other important features

Reply:
```suggestion
        return names.stream()
                .anyMatch(name -> file.getName()
                        .toLowerCase(Locale.ROOT)
                        .contains(name.toLowerCase(Locale.ROOT))) || names.isEmpty();
```
and of course same for the others

Reply:
> Streams are annoying. How exactly would the stream look? If it works like this I see no reason to change it and invest more time in a working feature before completing other important features

@kosch104 Streams werden sein Propra1 als way to go bezeichnet und man hätte es ja schon zuerst so machen können. Sie sind viel übersichtlicher und einfach Java Standard seit java 8.

Reply:
Hinzu kommt, dass die Komplexität verringern, indem man weniger Blöcke benutzt.

Reply:
there are even nice intellij features to convert loops to streams